overlord/snapstate/component.go

```diff
@@ -100,7 +106,11 @@ func InstallComponents(ctx context.Context, st *state.State, names []string, inf
 		}
 
 		compSetupIDs = append(compSetupIDs, componentTS.compSetupTaskID)
-		tss = append(tss, componentTS.taskSet())
+
+		ts := componentTS.taskSet()
+		ts.JoinLane(generateLane(st, opts))
```

Comment:
Does this imply that if transaction is per-snap then we also have a different lane per-component in each snap? Is that really what we want or maybe even if it is per-snap all components should by on the same lane as the snap?

Reply:
Good catch, I don't think that using anything besides one lane here makes sense.